stdlib/REPL/src/REPL.jl

```diff
@@ -510,6 +510,7 @@ function display(d::REPLDisplay, mime::MIME"text/plain", x)
             # this can override the :limit property set initially
             io = foldl(IOContext, d.repl.options.iocontext, init=io)
         end
+        io = IOContext(io, :displaysize => displaysize(outstream(d.repl)))
```

Comment:
What I don't get about this is that `io` is already initialized to `outstream(d.repl)`, so this should not be necessary. Is something else adding a `:displaysize` key here that's overriding it?

Reply:
I think the actual problem is in arrayshow.jl:
```
            sx = sprint(show, "text/plain", x, context=io, sizehint=0)
```
in print_matrix_row. It tries to inherit the context but the displaysize is not included.

Reply:
Okay, will try that then.